pkg/services/datasources/permissions/datasource_permissions.go

```diff
@@ -4,20 +4,26 @@ import (
 	"context"
 	"errors"
 
+	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/datasources"
 )
 
 var ErrNotImplemented = errors.New("not implemented")
 
 type DatasourcePermissionsService interface {
 	FilterDatasourcesBasedOnQueryPermissions(ctx context.Context, cmd *datasources.DatasourcesPermissionFilterQuery) error
+	FilterDatasourceUidsBasedOnQueryPermissions(ctx context.Context, user *models.SignedInUser, datasourceUids []string) ([]string, error)
 }
 
 // dummy method
 func (hs *OSSDatasourcePermissionsService) FilterDatasourcesBasedOnQueryPermissions(ctx context.Context, cmd *datasources.DatasourcesPermissionFilterQuery) error {
 	return ErrNotImplemented
 }
 
+func (hs *OSSDatasourcePermissionsService) FilterDatasourceUidsBasedOnQueryPermissions(ctx context.Context, user *models.SignedInUser, datasourceUids []string) ([]string, error) {
+	return nil, ErrNotImplemented
+}
+
```

Comment:
so this would only be enterprise query? 🤔

Reply:
yes, same as the existing one (https://github.com/grafana/grafana/pull/52706/files/bd0939a0ac008ddfa0bbbffd5bbca1c22a1720c1#diff-ec0367803710ed5316cf3f28d1201c31224567f0c4d5e0af71f2867dc81917e4R19)